### PR TITLE
[SPARK-30880][DOCS] Delete Sphinx Makefile cruft

### PR DESCRIPTION
**Pull request by nchammas**
Split off from #27534.

### What changes were proposed in this pull request?

This PR deletes some unused cruft from the Sphinx Makefiles.

### Why are the changes needed?

To remove dead code and the associated maintenance burden.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Tested locally by building the Python docs and reviewing them in my browser.

### CODE REVIEW

**Comment by nchammas** [comment]
I couldn't test the Windows Makefile. Perhaps @DavidToneian or @sarutak could take a look at the changes I made there?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Actually, @nchammas, what about we just switching it to minimized version of Makefile? (see https://github.com/sphinx-doc/sphinx/blob/af62fa61e6cbd88d0798963211e73e5ba0d55e6d/sphinx/templates/quickstart/Makefile.new_t,  https://github.com/databricks/koalas/blob/master/docs/Makefile and https://github.com/readthedocs/sphinx_rtd_theme/blob/master/docs/Makefile)?

File: python/docs/Makefile [hunk 1]
```diff
@@ -55,125 +25,3 @@ html:
 	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
```
Comment (srowen) [inline python/docs/Makefile]: So this is the only target we actually need? I'm Ok with it if we can't think of any reason to use the other targets. I assume this whole thing was a copy paste.

Comment (nchammas) [reply python/docs/Makefile]: As far as I can tell, `html` and `clean` are the only targets we use.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please don't reuse the same JIRA ID for different PRs, @nchammas .
- https://github.com/apache/spark/pull/27626 [SPARK-30731] Update deprecated Mkdocs option
- https://github.com/apache/spark/pull/27625 [SPARK-30731] Delete Sphinx Makefile cruft
- https://github.com/apache/spark/pull/27534 [SPARK-30731][DOCS] Refine workflow for building docs

**Comment by nchammas** [comment]
> Please don't reuse the same JIRA ID for different PRs, @nchammas .
> 
> * #27626 [SPARK-30731] Update deprecated Mkdocs option
> * #27625 [SPARK-30731] Delete Sphinx Makefile cruft
> * #27534 [SPARK-30731][DOCS] Refine workflow for building docs

Responded in two places:
* https://github.com/apache/spark/pull/27626#issuecomment-588391223
* https://github.com/apache/spark/pull/27534#issuecomment-588385183

**Comment by dongjoon-hyun** [comment]
Hi, @HyukjinKwon .
This should not land on `branch-2.4`.

**Comment by DavidToneian** [comment]
The makefile works for me on Windows. However, if one invokes `.\make.bat` without arguments, nothing happens. Given that all targets except `html` and `clean` have been removed, I find that it is more intuitive to accept `.\make.bat` (and `make` on other platforms, for that matter) to mean to build the documentation.

Should you disagree, I propose that `.\make.bat` / `make` without arguments should result in a message being printed that tells the user what targets there are.

**Comment by nchammas** [comment]
I've switched to more minimal Makefiles per @HyukjinKwon's suggestion. `make html` and `make clean` look good to me.

Calling `make` alone gives this:

```
$ make
Sphinx v2.3.1
Please use `make target' where target is one of
  html        to make standalone HTML files
  dirhtml     to make HTML files named index.html in directories
  singlehtml  to make a single large HTML file
  pickle      to make pickle files
  json        to make JSON files
  htmlhelp    to make HTML files and an HTML help project
  qthelp      to make HTML files and a qthelp project
  devhelp     to make HTML files and a Devhelp project
  epub        to make an epub
  latex       to make LaTeX files, you can set PAPER=a4 or PAPER=letter
  latexpdf    to make LaTeX and PDF files (default pdflatex)
  latexpdfja  to make LaTeX files and run them through platex/dvipdfmx
  text        to make text files
  man         to make manual pages
  texinfo     to make Texinfo files
  info        to make Texinfo files and run them through makeinfo
  gettext     to make PO message catalogs
  changes     to make an overview of all changed/added/deprecated items
  xml         to make Docutils-native XML files
  pseudoxml   to make pseudoxml-XML files for display purposes
  linkcheck   to check all external links for integrity
  doctest     to run all doctests embedded in the documentation (if enabled)
  coverage    to run coverage check of the documentation (if enabled)
```

@DavidToneian - I believe the Windows version should now do the same. Can you confirm?

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Yeah, let's dont port back to 2.4.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
@DavidToneian, do you mind if I ask to test the current `make.bat` out?

**Comment by DavidToneian** [comment]
The new `make2.bat` works for me on Windows, but I have two little issues that one might want to fix:

1. The `clean` target is not being advertised when running `make` / `.\make.bat`. I think it's a useful feature, but the user should be told about its existence.

2. On Windows, running `.\make.bat` without arguments prints the help message, which says to run `make`, when it probably should say `.\make.bat`. I know this output is due to Sphinx, but maybe one could `echo` a sentence before to make the user aware.

Thanks!

**Comment by HyukjinKwon** [comment]
@DavidToneian, thanks for testing it out. These files are actually templates from Sphinx so they might need to fix :D. I am pretty confident about this change because I am already using the exact same file.

@nchammas can you rebase? I think it's good to go.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Just for clarification, I didn't backport this PR because this change is potentially able to make diff in the output.

**Pull request closed without merging**